Mathlib/NumberTheory/Sylvester.lean

```diff
@@ -0,0 +1,97 @@
+/-
+Copyright (c) 2025 Walter Moreira, Joe Stubbs. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Walter Moreira, Joe Stubbs
+-/
+import Mathlib.Algebra.Order.Star.Basic
+import Mathlib.Analysis.Normed.Field.Lemmas
+
+/-!
+# Sylvester's sequence
+
+This file introduces the Sylvester's sequence.
+This is sequence [A000058](https://oeis.org/A000058) in [oeis].
+
+## Implementation notes
+
+We follow the presentantion from [Wikipedia](https://en.wikipedia.org/wiki/Sylvester%27s_sequence).
+
+## Main results
+
+- Basic facts: the first terms of the sequence are 2, 3, 7, 43.
+- `sylvester_prod_finset_add_one`: each term of the sequence is one plus the product of its
+  predecessors.
+- `sylvester_strictMono`: the sequence is strictly monotonic.
+- `sylvester_coprime`: Pairwise co-primality.
+
+## References
+
+* <https://en.wikipedia.org/wiki/Sylvester%27s_sequence>
+* [The On-Line Encyclopedia of Integer Sequences][oeis]
+-/
+
+open Nat
+
+/-- Sylvester's sequence: https://oeis.org/A000058. -/
+def sylvester : ℕ → ℕ
```

Comment:
I'm not sure if `sylvester` is a great name for this; https://en.wikipedia.org/wiki/James_Joseph_Sylvester has many things named after him, and it's not obvious to me that this is somehow most important.

`sylvesterSequence` would be ok, but maybe you or another reviewer have a better idea.

Reply:
I agree that `sylvester` is too vague. `sylvesterSequence` looks good to me, if it agrees with Mathlib conventions. We are open to any other name from the reviewers, as well.